lib/game_teams_manager.rb

```diff
@@ -33,12 +31,55 @@ def worst_coach(season)
     end[0]
   end
 
-  def game_team_results_by_season(season)
+  def most_accurate_team(season)
+    most_accurate_team_id = teams_shots_to_goals(season).max_by do |id, s_g|
+      s_g[:goals].to_f / s_g[:shots]
+    end[0]
+    find_team_by_team_id(most_accurate_team_id)
+  end
+
+  def least_accurate_team(season)
+    least_accurate_team_id = teams_shots_to_goals(season).min_by do |id, s_g|
+      s_g[:goals].to_f / s_g[:shots]
+    end[0]
+    find_team_by_team_id(least_accurate_team_id)
+  end
+
+  def most_tackles(season)
+    result_id = team_tackles(season).max_by {|id, tackle_count| tackle_count}
+    find_team_by_team_id(result_id[0])
+  end
+
+  def fewest_tackles(season)
+    result_id = team_tackles(season).min_by {|id, tackle_count| tackle_count}
+    find_team_by_team_id(result_id[0])
+  end
+
+  def game_teams_results_by_season(season)
     game_teams.find_all do |team_result|
       @tracker.find_game_ids_for_season(season).include? team_result.game_id
     end
   end
 
+# Need to test the following 3 methods?
+  def coaches_records(season)
+    gt_results = game_teams_results_by_season(season)
+    coach_record_start = initialize_coaches_records(gt_results)
+    total_record = add_wins_losses(gt_results, coach_record_start)
```

Comment:
Since the "total_record" variable isn't being used in this method, it looks like we don't need to store add_wins_losses in it!